internal/nostr/nostr.go

```diff
@@ -13,53 +13,42 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-const (
-	NIP_47_INFO_EVENT_KIND = 13194
-	NIP_47_REQUEST_KIND    = 23194
-	NIP_47_RESPONSE_KIND   = 23195
-)
-
-type WalletConnectInfo struct {
-	RelayURL     string
-	WalletPubkey string
-	Secret       string
-}
-
-type ErrorResponse struct {
-	Message string `json:"message"`
+type Config struct {
+	DefaultRelayURL string
 }
 
-type InfoRequest struct {
-	RelayURL     string `json:"relayUrl"`
-	WalletPubkey string `json:"walletPubkey"`
+type NostrService struct {
+	config      *Config
+	defaultRelay *nostr.Relay
 }
 
-type NIP47Request struct {
-	RelayURL     string       `json:"relayUrl"`
-	WalletPubkey string       `json:"walletPubkey"`
-	SignedEvent  *nostr.Event `json:"event"`
-	WebhookURL   string       `json:"webhookURL"`
-}
+func NewNostrService(config *Config) (*NostrService, error) {
+	logrus.Info("connecting to the relay...")
+	relay, err := nostr.RelayConnect(context.Background(), config.DefaultRelayURL)
```

Comment:
how do we we handle reconnecting here? 
what happens if the connection is lost?

Reply:
Whenever someone requests, we check if the connection exists, so this should be fine now!